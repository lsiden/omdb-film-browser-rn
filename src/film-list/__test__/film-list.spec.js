import React from "react"
import renderer from "react-native-test-utils"

import { filmList as FilmList } from "../"

jest.mock("components/pager")

const films = require("__test__/__fixture__/films.json").Search

jest.mock("film-list/film-list-item", () => "FilmListItem")

const defaultProps = () => {
  const dispatchViewDetail = jest.fn()
  return {
    films,
    dispatchViewDetail,
  }
}

const createWrapper = (props = {}) =>
  renderer(<FilmList {...{ ...defaultProps(), ...props }} />)

it("renders list of titles", () => {
  const wrapper = createWrapper()
  expect(wrapper.toJSON()).toMatchSnapshot()
})
