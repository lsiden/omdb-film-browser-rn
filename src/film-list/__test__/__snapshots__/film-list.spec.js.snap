// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders list of titles 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={undefined}
  >
    Found undefined results.
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#bbb",
          "borderTopWidth": 1,
          "marginTop": 20,
        },
        undefined,
      ]
    }
  >
    <FilmListItem
      filmSummary={
        Object {
          "Poster": "https://images-na.ssl-images-amazon.com/images/M/MV5BODAzMDgxMDc1MF5BMl5BanBnXkFtZTgwMTI0OTAzMjE@._V1_SX300.jpg",
          "Title": "The Hobbit: The Battle of the Five Armies",
          "Type": "movie",
          "Year": "2014",
          "imdbID": "tt2310332",
        }
      }
    />
    <FilmListItem
      filmSummary={
        Object {
          "Poster": "https://images-na.ssl-images-amazon.com/images/M/MV5BMDg2NzQwOGMtMGRkNC00YjAwLTg4NjgtZWQwYzljZmM1YzA4XkEyXkFqcGdeQXVyNTIzOTk5ODM@._V1_SX300.jpg",
          "Title": "Battle Los Angeles",
          "Type": "movie",
          "Year": "2011",
          "imdbID": "tt1217613",
        }
      }
    />
    <FilmListItem
      filmSummary={
        Object {
          "Poster": "https://images-na.ssl-images-amazon.com/images/M/MV5BMWI1YmQ0ZTctZWM2YS00YWE3LWE0NWUtNGE5YTFiNDlhOTY4L2ltYWdlXkEyXkFqcGdeQXVyMzM4MjM0Nzg@._V1_SX300.jpg",
          "Title": "Battle Royale",
          "Type": "movie",
          "Year": "2000",
          "imdbID": "tt0266308",
        }
      }
    />
    <FilmListItem
      filmSummary={
        Object {
          "Poster": "https://images-na.ssl-images-amazon.com/images/M/MV5BMGZjZDE2NjUtMzVkNy00N2UyLTg5MjYtYzY1ZmM1YWY3ZTZmXkEyXkFqcGdeQXVyNjk1Njg5NTA@._V1_SX300.jpg",
          "Title": "Night at the Museum: Battle of the Smithsonian",
          "Type": "movie",
          "Year": "2009",
          "imdbID": "tt1078912",
        }
      }
    />
    <FilmListItem
      filmSummary={
        Object {
          "Poster": "https://images-na.ssl-images-amazon.com/images/M/MV5BYzliNzI5NzgtNDZmNy00NGU4LTg2OTQtMDEzZTRiNDU3ZTI1XkEyXkFqcGdeQXVyMDUyOTUyNQ@@._V1_SX300.jpg",
          "Title": "The Battle of Algiers",
          "Type": "movie",
          "Year": "1966",
          "imdbID": "tt0058946",
        }
      }
    />
    <FilmListItem
      filmSummary={
        Object {
          "Poster": "https://images-na.ssl-images-amazon.com/images/M/MV5BZTljYmU2NTMtODhhNC00NjlhLWJhZTUtNDllODYyYWM4ZjA5XkEyXkFqcGdeQXVyNjM0ODk5NDY@._V1_SX300.jpg",
          "Title": "Battle of the Sexes",
          "Type": "movie",
          "Year": "2017",
          "imdbID": "tt4622512",
        }
      }
    />
    <FilmListItem
      filmSummary={
        Object {
          "Poster": "https://images-na.ssl-images-amazon.com/images/M/MV5BNWRlYjU3ODYtNzAzZS00YWE5LTkxZTEtYjM1YzIwZTlhMWQ2L2ltYWdlL2ltYWdlXkEyXkFqcGdeQXVyMTQxNzMzNDI@._V1_SX300.jpg",
          "Title": "Battle for the Planet of the Apes",
          "Type": "movie",
          "Year": "1973",
          "imdbID": "tt0069768",
        }
      }
    />
    <FilmListItem
      filmSummary={
        Object {
          "Poster": "https://images-na.ssl-images-amazon.com/images/M/MV5BMjE2MzUwODUxMl5BMl5BanBnXkFtZTgwMTI0NDkzMjE@._V1_SX300.jpg",
          "Title": "Dragon Ball Z: Battle of Gods",
          "Type": "movie",
          "Year": "2013",
          "imdbID": "tt2263944",
        }
      }
    />
    <FilmListItem
      filmSummary={
        Object {
          "Poster": "http://ia.media-imdb.com/images/M/MV5BMjE2MzUwODUxMl5BMl5BanBnXkFtZTgwMTI0NDkzMjE@._V1_SX300.jpg",
          "Title": "Dragon Ball Z: Battle of Gods",
          "Type": "movie",
          "Year": "2013",
          "imdbID": "tt2536562",
        }
      }
    />
    <FilmListItem
      filmSummary={
        Object {
          "Poster": "https://images-na.ssl-images-amazon.com/images/M/MV5BYjY1NmY1MzUtOTgwYy00Mzk0LThkNzgtZGRhNWU4NmFmYzMyL2ltYWdlXkEyXkFqcGdeQXVyNjQzNDI3NzY@._V1_SX300.jpg",
          "Title": "Battle of Britain",
          "Type": "movie",
          "Year": "1969",
          "imdbID": "tt0064072",
        }
      }
    />
  </View>
</View>
`;
